components/menu/menu.js

```diff
@@ -156,7 +160,7 @@ class Menu extends HierarchicalViewMixin(LitElement) {
 			items.unshift(returnItem);
 		}
 		return items.filter((item) => {
-			const role = item.role;
+			const role = item.role || item.getAttribute('role');
```

Comment:
After mucking around a bit, it doesn't seem like the ARIA role attribute reflects to a property -- although we've set up our Lit element so that the property reflects to the attribute. That's likely why you're seeing this. I wonder if we should just always use `getAttribute('role')` in this code then?

Reply:
In theory I think we could use just `getAttribute('role')` but with the template-menu demo (https://github.com/BrightspaceUI/menu/blob/master/demo/template-menu.html) `getAttribute('role')` only seems to be working on the first menu item, and the rest return `null`, where as `item.role` works (but then there are other cases like in `user-switcher` where `item.role` does not work and we need `item.getAttribute`). Not sure why that's happening - digging into it a bit.

Reply:
That's very weird... maybe a Polymer-y dom-repeat-y thing?

Reply:
Yeah I'm thinking so

Reply:
Adding a `requestAnimationFrame` inside of the `slotchange` event listener seems to fix that so we can just use `getAttribute('role')`

Reply:
Re the `requestAnimationFrame()`... maybe the role is not reflected to the attribute immediately upon `slotchange`... Lit needs a change to run its code to do that.  

You could also try `queueMicrotask()`, since that's what what Lit element uses for a bunch. You might not need to wait a full frame. It's not supported by Edge or IE11, but if you decide to try it and use it, there is a [super simple polyfil (few lines of code)](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/queueMicrotask#When_queueMicrotask_isnt_available) for it. It uses a Promise that resolves immediately (since that is microtask), and falls back to setTimeout (full task) when Promises aren't supported). If you do include it, I'd put it in our helpers next to `requestIdleCallback`. It's something I've been wanting to try and use, in effort to tighten up timing of some code I've written/writing.

Reply:
`queueMicrotask` actually works nicely! Thanks!

Reply:
So unfortunately after more cross-browser testing it seems this solution doesn't actually work for Safari. I have been able to get this solution working for all browsers if I replace `queueMicrotask` with `requestAnimationFrame` in `_onMenuItemsChanged`.

Reply:
Well it was a valiant effort nonetheless!